ispyb-ws/src/main/java/ispyb/ws/soap/mx/ToolsForBLSampleWebService.java

```diff
@@ -457,6 +457,30 @@ public String getStructuresByDataCollectionId(
 		}
 		return null;
 	}
+
+	/**
+	 * returns the pdb filePath for all crystals linked to a protein associated to a given dataCollectionId
+	 * 
+	 * @param code
+	 * @param number
+	 * @return
+	 * @throws Exception
+	 */
+	@WebMethod
+	@WebResult(name = "structures")
+	public String getProteinStructuresByDataCollectionId(
+	@WebParam(name = "dataCollectionId")Integer dataCollectionId) throws Exception {
+		try {
+			Structure3Service structure3service = (Structure3Service) ejb3ServiceLocator.getLocalService(Structure3Service.class);
+			List<Structure3VO> structures = structure3service.getProteinStructuresByDataCollectionId(dataCollectionId);
+			LOG.debug("getProteinStructuresByDataCollectionId. dataCollectionId=" + dataCollectionId + " structures=" + structures.size());
+			return convertStructuresToCSV(structures);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		return null;

```

Comment:
Do we want to return `null` ot to throw the error?

Reply:
I presume the null as the one they are using today returns null